include/zephyr/bluetooth/conn.h

```diff
@@ -390,7 +391,7 @@ struct bt_security_info {
 /** Connection Info Structure */
 struct bt_conn_info {
 	/** Connection Type. */
-	uint8_t type;
+	enum bt_conn_type type : 8;
```

Comment:
```suggestion
	enum bt_conn_type type;
```

Generally I don't think we want to use bitfields in public header files, as architecture support of these may vary.

Reply:
@jhedberg correct me if I am wrong here :)

Reply:
Although I am not strictly encouraging the use of bitfields, I decided to use them (as they *should* keep the old ABI and not inflate the necessary RAM needlessly) because a quick search revealed that Zephyr uses enum bitfields already in two USB-related public facing headers:

https://github.com/zephyrproject-rtos/zephyr/blob/6050a10f8b298fe6994bb47ab9775ba535a604dd/include/zephyr/drivers/usb/udc.h#L44

https://github.com/zephyrproject-rtos/zephyr/blob/6050a10f8b298fe6994bb47ab9775ba535a604dd/include/zephyr/drivers/usb_c/usbc_pd.h#L483

Reply:
The approach we've taken to compress enums into smaller space (in some few places where we thought it mattered), is to use `__packed` instead of bitfields, like here:
https://github.com/zephyrproject-rtos/zephyr/blob/90355f143f21d9ba985f62ea95d96ff83f6922ef/subsys/bluetooth/host/conn_internal.h#L14-L24

I think that'd be a better choice, since it ensures that the resulting size grows if we end up having more values than can fit a `uint8_t`.

Reply:
`__packed` is indeed a nice way to keep it small while still supporting expansions.